sources/typescript/src/adapters/kaltura-sequence-upload-request-adapter.ts

```diff
@@ -0,0 +1,164 @@
+                resumeAt: uploadedFileSize,
+            };
+
+            let activeAction: CancelableAction<any>;
+
+            const handleChunkUploadError = reason => {
+                activeAction = null;
+                reject(reason);
+            };
+
+            const handleChunkUploadSuccess = result => {
+                if (!data.enabled || data.finalChunk) {
+                    activeAction = null;
+                    this._handleFinalChunkResponse(request, result, reject, resolve);
+
+                } else {
+                    activeAction = this._chunkUpload(request, data).then(handleChunkUploadSuccess, handleChunkUploadError);
+                }
+            };
+
+            activeAction = this._chunkUpload(request, data)
+                .then(handleChunkUploadSuccess, handleChunkUploadError);
+
+
+            return () => {
+                if (activeAction) {
+                    activeAction.cancel();
+                    activeAction = null;
+                }
+            };
+        });
+    }
+
+    /**
+     * chunk upload for sequential chunks upload
+     * @param request
+     * @param uploadChunkData
+     * @private
+     */
+    private _chunkUpload(request: KalturaUploadRequest<any>, uploadChunkData: UploadBySequentialChunksData): CancelableAction<any> {
+        return new CancelableAction((resolve, reject) => {
+            const parameters = prepareParameters(request, this.clientOptions, this.defaultRequestOptions);
+
+            let isComplete = false;
+            const {propertyName, file} = request.getFileInfo();
+            let data = this._getFormData(propertyName, file.name, file);
+
+            let fileStart = 0;
+
+            if (uploadChunkData.enabled) {
+                let actualChunkFileSize: number = this._getChunkSize();
+
+                uploadChunkData.finalChunk = (file.size - uploadChunkData.resumeAt) <= actualChunkFileSize;
+
+                fileStart = uploadChunkData.resumeAt;
+                const fileEnd = uploadChunkData.finalChunk ? file.size : fileStart + actualChunkFileSize;
+
+                data = this._getFormData(propertyName, file.name, file.slice(fileStart, fileEnd, file.type));
+
+                parameters.resume = uploadChunkData.resume;
+                parameters.resumeAt = uploadChunkData.resumeAt;
+                parameters.finalChunk = uploadChunkData.finalChunk;
+            } else {
+                console.log(`chunk upload not supported by browser or by request. Uploading the file as-is`);
+            }
+
+            // const { service, action, ks, ...queryparams } = parameters;
+            const {service, action, ...queryparams} = parameters;
+            const endpointUrl = createEndpoint(request, this.clientOptions, service, action, queryparams);
+
+            // TODO should uncomment this part and verify that upload sends the ks as part of the body and not in the query param. see https://kaltura.atlassian.net/browse/KMCNG-2401
```

Comment:
??
i didnt see one in the parallel verison.

Reply:
this was there before me, I just copied that to a different file :) no idea what it means.